### PR TITLE
Add additional PgTextExpressionMethods

### PR DESCRIPTION
**Pull request by mattdowdell**
I found myself implementing these in my own project, and thought it might be useful to others (and means I can drop it from my code in the future). As a result, here's a PR with the following:

- Adds support for PostgreSQL's `SIMILAR TO` and `NOT SIMILAR TO`.
- Adds support for `ESCAPE` to PostgreSQL's `SIMILAR TO` and `NOT SIMILAR TO`.
- Added a summary of the changes to the changelog.

I think I've got everything necessary, but please let me know if there's more.

### CODE REVIEW

File: diesel/src/pg/expression/escape_expression_methods.rs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+impl<T, U> PgEscapeExpressionMethods for Grouped<NotSimilarTo<T, U>> {
+    type TextExpression = NotSimilarTo<T, U>;
+
+    fn escape(self, character: char) -> dsl::Escape<Self> {
+        Grouped(Escape::new(
+            self.0,
+            character.to_string().into_sql::<VarChar>(),
+        ))
+    }
+}
```
Comment (weiznich) [inline diesel/src/pg/expression/escape_expression_methods.rs:82]: Can you elaborate why this cannot use just the normal `ExscapeExpressionMethods` trait?

File: diesel/src/pg/expression/expression_methods.rs [hunk 2]
```diff
@@ -483,6 +483,62 @@ pub trait PgTextExpressionMethods: Expression + Sized {
...
+    /// #     let connection = establish_connection();
+    /// let starts_with_s = animals
+    ///     .select(species)
+    ///     .filter(name.similar_to("s%").or(species.similar_to("s%")))
+    ///     .get_results::<String>(&connection)?;
+    /// assert_eq!(vec!["spider"], starts_with_s);
+    /// #     Ok(())
+    /// # }
+    /// ```
+    fn similar_to<T>(selfm other: T) -> dsl::SimilarTo<Self, T>
```
Comment (weiznich) [inline diesel/src/pg/expression/expression_methods.rs:508]:
```suggestion
    fn similar_to<T>(self, other: T) -> dsl::SimilarTo<Self, T>
```

File: diesel/src/pg/expression/expression_methods.rs [hunk 3]
```diff
@@ -483,6 +483,62 @@ pub trait PgTextExpressionMethods: Expression + Sized {
...
+    /// #     let connection = establish_connection();
+    /// let doesnt_start_with_s = animals
+    ///     .select(species)
+    ///     .filter(name.not_similar_to("s%").and(species.not_similar_to("s%")))
+    ///     .get_results::<String>(&connection)?;
+    /// assert_eq!(vec!["dog"], doesnt_start_with_s);
+    /// #     Ok(())
+    /// # }
+    /// ```
+    fn similar_to<T>(selfm other: T) -> dsl::SimilarTo<Self, T>
```
Comment (weiznich) [inline diesel/src/pg/expression/expression_methods.rs:536]:
```suggestion
    fn similar_to<T>(self, other: T) -> dsl::SimilarTo<Self, T>
```

File: diesel/src/pg/expression/escape_expression_methods.rs (hunk 1 above)
Comment (mattdowdell) [reply diesel/src/pg/expression/escape_expression_methods.rs:82]: Quite simply, it didn't occur to me. I'll switch over to that :)

File: diesel/src/pg/expression/helper_types.rs [hunk 4]
```diff
@@ -8,6 +8,12 @@ pub type ILike<Lhs, Rhs> = Grouped<super::operators::ILike<Lhs, AsExprOf<Rhs, Va
 /// The return type of `lhs.not_ilike(rhs)`
 pub type NotILike<Lhs, Rhs> = Grouped<super::operators::NotILike<Lhs, AsExprOf<Rhs, VarChar>>>;
 
+/// The return type of `lhs.ilike(rhs)`
+pub type SimilarTo<Lhs, Rhs> = Grouped<super::operators::SimilarTo<Lhs, AsExprOf<Rhs, VarChar>>>;
+
+/// The return type of `lhs.not_ilike(rhs)`
+pub type NotSimilarTo<Lhs, Rhs> = Grouped<super::operators::NotSimilarTo<Lhs, AsExprOf<Rhs, VarChar>>>;
```
Comment (mattdowdell) [inline diesel/src/pg/expression/helper_types.rs:15]: not to self, update docs here

File: diesel/src/pg/expression/helper_types.rs [hunk 5]
```diff
@@ -8,6 +8,12 @@ pub type ILike<Lhs, Rhs> = Grouped<super::operators::ILike<Lhs, AsExprOf<Rhs, Va
 /// The return type of `lhs.not_ilike(rhs)`
 pub type NotILike<Lhs, Rhs> = Grouped<super::operators::NotILike<Lhs, AsExprOf<Rhs, VarChar>>>;
 
+/// The return type of `lhs.ilike(rhs)`
```
Comment (weiznich) [inline diesel/src/pg/expression/helper_types.rs:11]:
```suggestion
/// The return type of `lhs.similar_to(rhs)`
```

File: diesel/src/pg/expression/helper_types.rs [hunk 6]
```diff
@@ -8,6 +8,12 @@ pub type ILike<Lhs, Rhs> = Grouped<super::operators::ILike<Lhs, AsExprOf<Rhs, Va
...
+pub type SimilarTo<Lhs, Rhs> = Grouped<super::operators::SimilarTo<Lhs, AsExprOf<Rhs, VarChar>>>;
+
+/// The return type of `lhs.not_ilike(rhs)`
```
Comment (weiznich) [inline diesel/src/pg/expression/helper_types.rs:14]:
```suggestion
/// The return type of `lhs.not_similar_to(rhs)`
```

**Comment by mattdowdell** [comment]
@weiznich Thanks for the super quick reviews!

I think I've got to the point where CI will pass (the last run was a fmt issue and a transient(?) issue on MacOS) but it apparently declined to run this time. Is there anything else you'd like to see here?

**Review by weiznich — APPROVED** [review]
Looks good as soon as the CI passes (the MacOS failure is fine).
Thanks for submitting this PR :+1:

**Review by TaKO8Ki — APPROVED** [review]

**Pull request merged**